app/Shell/Environment.php

```diff
@@ -11,10 +13,34 @@ public function __construct(Shell $shell)
         $this->shell = $shell;
     }
 
+    private function isLinuxOs()
+    {
+        return PHP_OS_FAMILY === 'Linux';
+    }
+
     public function portIsAvailable($port): bool
     {
+        $isLinux = $this->isLinuxOs();
+        $netstatCmd = 'netstat';
+        $portText = "\.{$port}\s"; // mac default port behavior 127.0.0.1.3306 for mysql
+
+        // if its linux we need to check WSL also,
+        // because WSL runs docker from Windows host
+        // and it uses netstat.exe from Windows
+        // if we don't use this, WSL won't find ports conflict
+        if ($isLinux) {
+            $portText = "\:{$port}\s"; // win/linux default port behavior 127.0.0.1:3306 for mysql
+
+            $linuxVersion = $this->shell->execQuietly('cat /proc/version');
+            $isWSL = Str::contains($linuxVersion->getOutput(), 'microsoft');
+
+            if ($isWSL) {
+                $netstatCmd = 'netstat.exe';
+            }
+        }
+
         // Check to see if the system is running a service with the desired port
-        $process = $this->shell->execQuietly("netstat -vanp tcp | grep '\.{$port}\s' | grep -v 'TIME_WAIT' | grep -v 'CLOSE_WAIT' | grep -v 'FIN_WAIT'");
+        $process = $this->shell->execQuietly("{$netstatCmd} -vanp tcp | grep '{$portText}' | grep -v 'TIME_WAIT' | grep -v 'CLOSE_WAIT' | grep -v 'FIN_WAIT'");
```

Comment:
I super appreciate the help with cross-platform stuff. Like before, it's much easier to review and test a PR when it has a single feature. Let's leave this in and I'll run tests on WSL.